src/pages/home/MainView.js

```diff
@@ -20,21 +20,30 @@ const propTypes = {
     reports: PropTypes.objectOf(PropTypes.shape({
         reportID: PropTypes.number,
     })),
+
+    currentlyViewedReportID: PropTypes.string,
+
+    currentURL: PropTypes.string,
 };
 
 const defaultProps = {
     reports: {},
+    currentlyViewedReportID: '',
+    currentURL: '',
 };
 
 class MainView extends Component {
     render() {
-        const reportIDInUrl = parseInt(this.props.match.params.reportID, 10);
+        let activeReportID = parseInt(this.props.match.params.reportID, 10);
```

Comment:
I'm curious what would happen if we simply stopped relying on the match params here. IIRC we should be updating the `currentlyViewedReportID` when we hit one of these routes.

Reply:
By that you mean just doing
```JS
activeReportID = parseInt(this.props.currentlyViewedReportID, 10);
```

always, right?

Reply:
Yeah exactly. I'm trying to think of a time when the reportID in the URL wouldn't be the `currentlyViewedReportID` and not sure there is one. If we don't need it then the logic in this file becomes a little simpler and we can stop making an exception for the settings route.